File: services/gocore/internal/service/grpc/cpu_heavy_service.go
// Package servicegrpc
package servicegrpc

import (
	"context"
	pb "gocore/pkg/proto"
	"gocore/pkg/utils"
	"log"
)

// CPUHeavyServiceServer CpuHeavyService의 요청을 처리하는 구조체
type CPUHeavyServiceServer struct {
	pb.UnimplementedCpuHeavyServiceServer
}

// ExecuteHeavyCalculation CPU-Heavy 연산을 수행하는 핸들러 함수
func (s *CPUHeavyServiceServer) ExecuteHeavyCalculation(ctx context.Context, req *pb.HeavyCalculationRequest) (*pb.HeavyCalculationResponse, error) {

	result := utils.HeavyCalculation(req.GetComplexityLevel())

	log.Printf("INFO: Calculation completed. Checksum: %d", result)

	return &pb.HeavyCalculationResponse{
		Success:        true,
		ResultChecksum: result,
	}, nil
}

File: services/flask-gateway/services/internal/cpu_heavy_service.py
from utils.cpu_heavy import heavy_calculation

def cpu_heavy_service(complexity_level: int) -> int:
    """복잡한 연산을 하는 함수를 호출합니다.

    Args:
        complexity_level (int): 연산 난이도

    Returns:
        int: 연산 결과
    """
    result = heavy_calculation(complexity_level=complexity_level)
    return result